assets/js/modules/forms.js

```diff
@@ -30,6 +31,28 @@ function handleAbandonmentMessage(formEl) {
     });
 }
 
+/*
+ * Warns the user before leaving a page if the form's state is different from when the page was loaded.
+ *
+ * Caveats: form must have the class below and the field must lose focus to trigger the change() event.
+ * */
+function warnOnUnsavedChanges() {
+    let formHasChanged = false;
+    $('.js-form-warn-unsaved').each(function() {
+        const $form = $(this);
+        const initialState = $form.serialize();
+        $(':input', $form).change(function() {
+            const newState = $form.serialize();
+            formHasChanged = newState !== initialState;
```

Comment:
Do we need the `formHasChanged` variable, could we inline the expression into the if statement?

Reply:
Done!